protocol/messenger_communities.go

```diff
@@ -1966,7 +1966,7 @@ func (m *Messenger) CreateCommunity(request *requests.CreateCommunity, createDef
 	return response, nil
 }
 
-func (m *Messenger) CreateCommunityTokenPermission(request *requests.CreateCommunityTokenPermission) (*MessengerResponse, error) {
+func (m *Messenger) CreateCommunityTokenPermissionWithSkipParam(request *requests.CreateCommunityTokenPermission, skipReevaluation bool) (*MessengerResponse, error) {
```

Comment:
I'm not sure that bool flags look right here. In my understanding, we are adding CommunityDescription with the permissions and saving it to the DB. After it is saved, we need to do a permission verification based on the permissions in the CommunityDescription

What could be wrong:
1.  CommunityDescription was not saved
2. We pass the old community description, which does not contains new Permissions

Reply:
I only use that param in tests, though I get that it's not that clear and it's not clear that it shouldn't be used normally.

I can try to change the tests to wait for the reevaluation.

Reply:
Okay, checked the test and I think I found the issue
When you are calling `s.advertiseCommunityTo(community, s.bob)` - you are sending an outdated community. You must request community from the DB or extract it from the last response

Reply:
nah... check where it is used and that's not the case :(

Reply:
Alright, I used `waitOnCommunityPermissionCreated` after the permission is created to wait for the re-evaluation.

It seems to work well and is way simpler.

You can review again